src/character-card-parser.js

```diff
@@ -23,9 +23,24 @@ const write = (image, data) => {
         }
     }
 
-    // Add new chunks before the IEND chunk
+    // Add new v2 chunk before the IEND chunk
     const base64EncodedData = Buffer.from(data, 'utf8').toString('base64');
     chunks.splice(-1, 0, PNGtext.encode('chara', base64EncodedData));
+
+    // Try adding v3 chunk before the IEND chunk
+    try {
+        //change v2 format to v3
+        const v3Data = JSON.parse(data);
+        v3Data.spec = 'chara_card_v3'
+        v3Data.spec_version = '3.0'
+        if(v3Data.data && !v3Data.data.group_only_greetings){
```

Comment:
I'd rather move this to `/src/endpoints/characters.js` to set this array in the data for all specs.

Reply:
To be specific, here:

https://github.com/SillyTavern/SillyTavern/blob/release/src/endpoints/characters.js#L348

Reply:
Wouldn't that break the V2 spec? In the V2 spec, the `spec` is `chara_card_v2`, and when it becomes `chara_card_v3` to match the v3 spec, it becomes an invalid V2 spec.

Reply:
No, don't touch the spec field, just the `group_only_greetings` field.